invoices/invoiceregistry.go

```diff
@@ -453,6 +450,9 @@ func (i *InvoiceRegistry) NotifyExitHopHtlc(rHash lntypes.Hash,
 		customRecords:        payload.CustomRecords(),
 	}
 
+	i.Lock()
```

Comment:
Would be nice to have rationale for this in the commit message body.

Reply:
Added